content/criando-novos-comandos-no-django-admin.md

```diff
@@ -0,0 +1,395 @@
+
+### hello.py
+
+```python
+from django.core.management.base import BaseCommand, CommandError
+# minimalista
+class Command(BaseCommand):
+    help = 'Print hello world'
+
+    def handle(self, **options):
+        print('Hello World')
+```
+
+**Uso**
+
+```bash
+$ ./manage.py hello
+```
+
+### initdata.py
+
+**Objetivo**: Obter alguns filmes de uma api e salvar os dados no banco.
+
+**api**: [omdbapi.com][3]
+
+**models.py**
+
+```python
+from django.db import models
+
+class Movie(models.Model):
+    title = models.CharField(u'título', max_length=100)
+    year = models.PositiveIntegerField('ano', null=True, blank=True)
+    released = models.CharField(u'lançamento', max_length=100, null=True, blank=True)
+    director = models.CharField('diretor', max_length=100, null=True, blank=True)
+    actors = models.CharField('atores', max_length=100, null=True, blank=True)
+    poster = models.URLField('poster', null=True, blank=True)
+    imdbRating = models.DecimalField(max_digits=6, decimal_places=2, null=True, blank=True)
+    imdbVotes = models.PositiveIntegerField(null=True, blank=True)
+    imdbID = models.CharField(max_length=50, null=True, blank=True)
+
+    class Meta:
+        ordering = ['title']
+        verbose_name = 'filme'
+        verbose_name_plural = 'filmes'
+
+    def __str__(self):
+        return self.title
+```
+
+Não se esqueça de fazer
+
+```bash
+./manage.py makemigrations
+./manage.py migrate
+```
+
+**admin.py**
+
+Vamos visualizar pelo admin.
+
+```python
+from django.contrib import admin
+from core.models import Movie
+
+admin.site.register(Movie)
+```
+
+Instale o `requests`
+
+```bash
+pip install requests
+```
+
+**initdata.py**
+
+```python
+import random
+import string
+import requests
+from django.core.management.base import BaseCommand, CommandError
+from django.core.exceptions import ValidationError
+from optparse import make_option
+from core.models import Movie
+
+
+class Command(BaseCommand):
+    help = 'Faz o crawler numa api de filmes e retorna os dados.\n \
+    Uso: ./manage.py initdata\n \
+    ou: ./manage.py initdata -m 20\n \
+    ou: ./manage.py initdata -m 20 -y 2015'
+    option_list = BaseCommand.option_list + (
+        make_option('--movies', '-m',
+                    dest='movies',
+                    default=10,
+                    help='Define a quantidade de filmes a ser inserido.'),
+        make_option('--year', '-y',
+                    dest='year',
+                    action='store',
+                    default=None,
+                    help='Define o ano de lançamento do filme.'),
+    )
+
+    def print_red(self, name):
+        print("\033[91m {}\033[00m".format(name))
+
+    def get_html(self, year):
+        '''
+        Le os dados na api http://www.omdbapi.com/ de forma aleatoria
+        e escolhe um filme buscando por 2 letras
+        '''
+
+        ''' Escolhe duas letras aleatoriamente '''
+        c = ''.join(random.choice(string.ascii_lowercase)for _ in range(2))
```

Comment:
@rg3915 para clarear o entendimento do leitor, seria legal nomear melhor suas variaveis

 `Explicit is better than implicit.`